### DISCUSSION TITLE
How to control if jobs and triggers are working properly?

### DISCUSSION THREAD

**Discussion by hahyes** [quartznet/quartznet#1173] [Q&A]
Hi there,

I'm making first steps into Quartz and something worries me. Is there a way to report progress like in `BackgroundWorker` class? Currently I'm using `BackgroundWorker` and custom scheduler to make some recurring jobs. I'm using `.ReportProgress` and event to log jobs progressing into database. Is there something similar to that? I suppose I could use LogProvider for that, but I'm curious if I could do this better way.

Additional question: is `Listener` in Quartz equivalent of firing events in `BackgroundWorker`? I'm not sure I understand documentation correctly.

Cheers.

**Comment by lahma** [comment]
There's not built-in. You could inject a a service if you want to and then report using it while job is running. context has FireInstanceId which is unique and can be used as key for running job. Also a listener could ensure that initial state and final state is written to some storage that you use to maintain this information.
